synapse/handlers/federation.py

```diff
@@ -2498,6 +2504,7 @@ async def _check_for_soft_fail(
             event_auth.check(room_version_obj, event, auth_events=current_auth_events)
         except AuthError as e:
             logger.warning("Soft-failing %r because %s", event, e)
```

Comment:
> For the future:
> 
> If we want to dig deeper into what rooms are soft_failing messages, we could use the Elasticsearch logs and some aggregations (Kibana or raw ES queries). We can add a few fields to the Elasticsearch mapping, room_id, mxid, event_id which should handle the high cardinality(lots of different values) just fine.
> 
> It looks like we use Logstash or Filebeat and I assume there is a way we can parse another field out besides message? Maybe https://stackoverflow.com/q/40460830/796832
> 
> *-- https://github.com/matrix-org/synapse/pull/10156#issuecomment-859287143*


To document my additional findings here;

We can already use `logger.info('foo', extra={ "foo": "bar"}` as shown in [`tests/logging/test_terse_json.py#L64-L74`](https://github.com/matrix-org/synapse/blob/13577aa55ebe6087e8b813c0643bbb53148e9510/tests/logging/test_terse_json.py#L64-L74) to add extra fields to the structured logging.

 - Docs on structured logging in Synapse: https://github.com/matrix-org/synapse/blob/develop/docs/structured_logging.md
 - Improve structured logging: https://github.com/matrix-org/synapse/issues/8683
    - Record more information into structured logs: https://github.com/matrix-org/synapse/pull/9654
 - Use structured logging for EMS: https://github.com/matrix-org/matrix-hosted/issues/1229
 - Use structured logging for `matrix.org`: https://github.com/matrix-org/internal-config/issues/889
 - [Allow simultaneously enabling structured logging and "standard" logging](https://github.com/matrix-org/synapse/issues/8588) which was implemented in https://github.com/matrix-org/synapse/pull/8607
 - List of issues with the `z-logging` label: https://github.com/matrix-org/synapse/labels/z-logging

Reply:
Created https://github.com/matrix-org/synapse/pull/10168 to add these fields